### PR TITLE
Added conftest type hints

### PR DESCRIPTION
**Pull request by sudosubin**
I'm not sure this test case would be helpful. Feel free to comment.

- Add small type hints for `conftest.py`
- ~~Add a logging module test case (`test_make_console_logger`)~~ -> Reverted
- Add pytest configuration (`asyncio_mode = auto`) to suppress `DeprecationWarning` when running pytest

  ```sh
  pytest_asyncio/plugin.py:191: DeprecationWarning: The 'asyncio_mode' default value will change to 'strict' in future, please explicitly use 'asyncio_mode=strict' or 'asyncio_mode=auto' in pytest configuration file.
      config.issue_config_time_warning(LEGACY_MODE, stacklevel=2)
  ```

### CODE REVIEW

File: tests/test_logging.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+    with caplog.at_level(logging.INFO, logger="notion_client"):
+        logger.info("TEST_LOGGER_INFO_MESSAGE")
+        assert caplog.records[0].message == "TEST_LOGGER_INFO_MESSAGE"
+        caplog.clear()
+
+    with caplog.at_level(logging.DEBUG, logger="notion_client"):
+        logger.debug("TEST_LOGGER_DEBUG_MESSAGE")
+        assert caplog.records[0].message == "TEST_LOGGER_DEBUG_MESSAGE"
+        caplog.clear()
```
Comment (ramnes) [inline tests/test_logging.py]: Can we test our actual logging calls rather than the logging library itself? 😅

**Review by ramnes — CHANGES_REQUESTED** [review]
I'm good with the first two commits, feel free to throw them into another PR if you want them merged quickly – or see my comment for the third commit. :)

File: tests/test_logging.py (hunk 1 above)
Comment (sudosubin) [reply tests/test_logging.py]: I agree with your comment :)
It seems better to have `Client.request`, `Async.request` methods tested in this way.

Comment (sudosubin) [reply tests/test_logging.py]: I will revert the third commit from this PR, and will open new PR for logging test :)

**Comment by ramnes** [comment]
Thanks!

**Pull request merged**